app/services/technical_metadata_generator.rb

```diff
@@ -138,7 +138,8 @@ def generate_metadata_for_mimetype(mimetype, filepath)
   # rubocop:enable Metrics/MethodLength
 
   def filename_for(filepath)
-    ::File.basename(filepath)
+    # For example, /dor/workspace/cy/002/pz/9778/cy002pz9778/content/dir/file.txt
+    filepath.gsub(%r{^.+?/content/}, '')
```

Comment:
Not important right now, but I'm wondering if a regex-based approach to parsing the path out of the string is more brittle than, say, something like:

```ruby
    druid_obj = DruidTools::Druid.new(druid, '/dor/workspace') # not sure if the workspace dir already lives in config somewhere for techmd, but it could?
    filepath.slice!("#{druid_obj.content_dir(false)/")
    filepath
```

For now, and possibly also later, the regex is completely fine. 😄 Food for thought.

Reply:
So TechMd was designed to be DOR agnostic, which is why file URIs are submitted in the API. Both the regex and using DruidTools violates this.

For actual HFS implementation, we can consider either (1) changing the API to accept filenames instead of file URIs (which might include a path) and then using DruidTools or (2) adding a basepath parameter to the API (which is used to determine which part of the filepath to remove).

Reply:
See #398